test/check-machines-consoles

```diff
@@ -744,6 +745,102 @@ fullscreen=0
         b.wait_visible(".vm-console-vnc canvas")
         self.assertIn("password=on", m.execute(f"grep ^-vnc /var/log/libvirt/qemu/{name}.log"))
 
+        # Change global password, but without restarting the VM. This
+        # means that Cockpit will use the new password when
+        # connecting, but the VM still expects the old.
+        m.execute("sed -i -e 's/vnc_password = \"barfoo\"/vnc_password = \"foobar\"/' /etc/libvirt/qemu.conf")
+
+        # We are still connected, but reconnecting will fail.
+
+        b.click(".consoles-card button:contains(Disconnect)")
+        b.wait_in_text(".consoles-card", "Disconnected")
+
+        # Quite often the reconnection here fails with code 1006
+        # (Abnormal Closure), but retrying will give us the expected
+        # error message, usually on the second try.
+
+        b.click(".consoles-card button:contains(Connect)")
+        for _ in range(5):
+            try:
+                b.wait_in_text(".consoles-card", "Authentication failed")
+                break
+            except testlib.Error:
+                if "Failed to connect" in b.text(".consoles-card"):
+                    print("Unexpected error message. Retrying...")
+                    b.click(".consoles-card button:contains(Retry)")
+                else:
+                    raise
+
+        b.wait_in_text(".consoles-card", "Authentication failed")
+
+    @testlib.skipImage("cockpit-ws too old",
+                       "rhel-9-*", "centos-9-*", "ubuntu-*", "debian-trixie", "opensuse-tumbleweed")
+    def testNonSharedExternal(self):
+        b = self.browser
+
+        name = "subVmTest1"
+        self.createVm(name, graphics="vnc")
+
+        self.login_and_go("/machines")
+        self.waitPageInit()
+        self.waitVmRow(name)
+        self.goToVmPage(name)
+
+        # VNC viewer should be connected
+        b.wait_visible(".vm-console-vnc canvas")
+
+        # Start external non-shared viewer.  We use Cockpit itself, in
+        # "Remote resizing" mode.
+        b2 = self.new_browser()
+        cookie = b.cookie("cockpit")
+        b2.open(f"/cockpit/@localhost/machines/index.html#/vm/vnc?name={name}&connection=system", cookie=cookie)
+        with b2.wait_timeout(60):
```

Comment:
This is a bit strange -- perhaps wait for the page initialization first (with a longer timeout), and then "normally" do the dropdown thing? (Not a veto, just thinking aloud)

Reply:
tbh, I copy/pasted the big timeout without thinking. I'll check this out.

Reply:
Right, in other places 60 seconds account for the time it takes the guest to properly set up the size of its VNC framebuffer. This happens sometime during boot.  Here we have to wait for the page to initialize itself, which is unfortunately slow.

We have `self.waitPageInit` for that, but that only works for `self.browser`.

I left this but added a comment with a link to https://issues.redhat.com/browse/COCKPIT-1302